src/System.Device.Gpio/System/Device/Gpio/PinValuePair.cs

```diff
@@ -17,5 +17,47 @@ public PinValuePair(int pin, PinValue value)
             PinNumber = pin;
             PinValue = value;
         }
+
+        /// <summary>
+        /// Deconstructor for convenience. Allows using as a "return Tuple".
+        /// </summary>
+        public void Deconstruct(out int pinNumber, out PinValue pinValue)
+        {
+            pinNumber = PinNumber;
+            pinValue = PinValue;
+        }
+
+        /// <summary>
+        /// Returns a bit mask of designated pins. Supports pin numbers from 0 to 31.
+        /// </summary>
+        /// <remarks>
+        /// On some devices (such as GPIO extenders) pin numbers are low and often consecutive. This
+        /// is a helper to turn sets of pins into bits/bitmasks for storage and/or sending over the
+        /// wire to devices that want pin numbers in a bit format.
+        /// </remarks>
+        /// <param name="pinValues">A set of pin value pairs.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown if given a negative pin or pin number of 32 or higher.
```

Comment:
Could this bite us later?  For example, The [CY8C95xx](https://www.cypress.com/file/37971/download) I plan to work on soon has 20/40/60 I/Os.

Reply:
It doesn't prevent us from adding another helper. I did find out were not allowed to use tuples in public API so I'll have to change this. `BitVector32` is useful, but we don't have a 64 bit equivalent.

I discussed it a bit and I think the right answer (that will get past the review) is to add a `PinVector32` and `PinVector64` struct and respective helpers here. We can add deconstructors to those which will allow using them tuple style if desired without actually exposing tuples (verboten).